code/game/objects/effects/effect_system.dm

```diff
@@ -42,8 +42,7 @@ would spawn and follow the beaker, even if it is carried or thrown.
 /obj/effect/proc/delete()
 	loc = null
 	if(reagents)
-		reagents.my_atom = null
-		reagents.delete()
+		qdel(reagents)

```

Comment:
This whole proc should get removed entirely, but in the meantime, is there any reason it can't just qdel(src)? Do effects get reused or something?